addons/survey/tests/test_survey_invite.py

```diff
@@ -258,3 +258,25 @@ def test_survey_invite_token_by_email_nosignup(self):
         answers = self.env['survey.user_input'].search([('survey_id', '=', self.survey.id)])
         self.assertEqual(len(answers), 1)
         self.assertEqual(answers.partner_id.display_name, first_partner.display_name)
+
+    def test_survey_with_template_attachment(self):
+        """
+        Ensure a survey email will send an attachment that has been added to its respective template
+        """
+        mail_template = self.env['mail.template'].create({
+            'name': 'test mail template',
+            'attachment_ids': [Command.create({
+                'name': 'some_attachment.pdf',
+                'res_model': 'mail.template',
+                'datas': 'test',
+                'type': 'binary',
+            })],
+        })
+
+        action = self.survey.action_send_survey()
+        invite_form = Form(self.env[action['res_model']].with_context(action['context']))
+        invite_form.template_id = mail_template
+        self.assertEqual(
+            self.env['ir.attachment'].search([('id', '=', invite_form._values['attachment_ids'][0][2][0])], limit=1).id,
```

Comment:
Seems a bit convoluted ^^ for 2many in forms isn't it a "records" variable or something like that ? I don't remember the actual name but there is something simpler :D